### PR TITLE
ENH roll our own label binarizer to avoid sklearn edge cases

### PR DESCRIPTION
**Pull request by beckermr**
This PR rolls our own simple label binarizer (based on the sklearn one) in order to avoid weird edge cases we found when using theirs.

### CODE REVIEW

**Comment by beckermr** [comment]
FWIW, the transformer tests cover all of these cases, but I will add the tests.

File: civismlext/preprocessing.py [hunk 1]
```diff
@@ -7,9 +7,69 @@
...
+        Unique set of classes.
+
+    Returns
+    -------
+    yexp : numpy array of shape [n_samples, n_classes]
+        Categorically expanded data.
+    """
+    # Preprocess data to array format.
+    if not isinstance(y, list):
+        # XXX Workaround that will be removed when list of list format is
```
[the remark this answers is not in the document]
Comment (beckermr) [reply civismlext/preprocessing.py]: From sklearn. left it since I could not figure it out.

**Comment by beckermr** [comment]
@elsander I got all of the tests. Back to you!

File: civismlext/preprocessing.py [hunk 2]
```diff
@@ -7,9 +7,97 @@
...
+           [0, 1, 0]])
+    >>> _label_binarize(['a', 'c', 'b'], classes=['a', 'b', 'c', 'd'])
+    array([[1, 0, 0, 0],
+           [0, 0, 1, 0],
+           [0, 1, 0, 0]])
+    """
+    # Preprocess data to array format.
+    if not isinstance(y, list):
+        # the comment below is from sklearn v0.19
+        # XXX Workaround that will be removed when list of list format is
```
Comment (elsander) [inline civismlext/preprocessing.py]: (lol)
